compiler/next/lib/uast/Coforall.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/uast/Coforall.h"
+
+#include "chpl/uast/Builder.h"
+
+namespace chpl {
+namespace uast {
+
+
+owned<Coforall> Coforall::build(Builder* builder, Location loc,
+                                owned<Decl> index,
+                                owned<Expression> iterand,
+                                owned<WithClause> withClause,
+                                ASTList stmts,
+                                bool usesImplicitBlock) {
+  assert(iterand.get() != nullptr);
+
+  ASTList lst;
+  int8_t indexChildNum = -1;
```

Comment:
These `int8_t`s are a little worrisome and made me take a closer look because it's so easy to overflow 7 bits. I see that they stay small and came from `IndexableLoop`. But it made me do a double take when I saw them.

Reply:
Yeah the reason why we use `int8_t` here is because we know these indices will always be small (these things are always stored at the front of the children list in fixed positions).

The `loopBodyChildNum` and `numLoopBodyStmts` fields of `Loop` both use `int` because they can appear anywhere in the children list.